### DISCUSSION TITLE
How do I set the font for math?

### DISCUSSION THREAD

**Discussion by ararunaufc** [typst/typst#567] [Q&A] [answered]
I tried using `show math.text: set text(font: ...)` but it changes the font for the whole document. Also `set math.text(font: ...)` doesn't seem to make any difference...

I've seen it already in Discord and somewhere in the docs (maybe?), but I can't find it anymore...

Thanks.

**Comment by laurmaedje** [comment, marked as answer]
Almost. It's `#show math.equation: set text(font: ...)`.

**Reply by ararunaufc** [reply]
Oooohhh...! 😅 
🔥

**Comment by spikespaz** [comment]
I don't understand why this is the case, why does `math.text` change the font for the whole document?

**Reply by laurmaedje** [reply]
It's a bit confusing, but `math.text` and `text` is the exact same thing. The `text` function is just redefined in the math scope so that it is available in equations (e.g. `$ text(#red, 1 + 2) $`).

**Reply by spikespaz** [reply]
@laurmaedje Interesting. I would have thought assigning a function in another scope would `Clone` the function's context so that any changes to the new context would only apply within said scope, without side effects. Do you have any more information about how this behavior is intended to work? Is it intentional?

**Reply by laurmaedje** [reply]
`#set math.text(..)` and `#set text(..)` and `#let t = text; #set t(..)` are all 100% equivalent. Writing `#set math.text(..)` is possible because `text` is part of the `math` module (so that it is available in equations without use of a hash), but it's confusing and doesn't ever make sense.

**Reply by spikespaz** [reply]
@laurmaedje Yes, I understand. I am asking why it is that way.
